proxy/proxy_handlers.go

```diff
@@ -0,0 +1,87 @@
+package proxy
+
+import (
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/webhooktunnel/wsmux"
+)
+
+func (p *proxy) handler(w http.ResponseWriter, r *http.Request) {
+	// register a worker
+	log.Printf("new request: %s", r.URL.Path)
+	if strings.HasPrefix(r.URL.Path, "/register") {
+		p.register(w, r)
+		return
+	}
+	log.Printf("serving content:")
+	p.serve(w, r)
+}
+
+/*
+register is used to connect a worker to the proxy so that it can start serving API endpoints.
+The request must be a websocket upgrade request with a header field containing x-worker-id.
+The request is validated by the proxy and the http connection is upgraded to websocket.
+*/
+func (p *proxy) register(w http.ResponseWriter, r *http.Request) {
+	if !websocket.IsWebSocketUpgrade(r) {
+		http.NotFound(w, r)
+		return
+	}
+	if err := p.validateRequest(r); err != nil {
+		http.Error(w, "invalid request", 401)
+		return
+	}
+
+	conn, err := p.upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		panic(err)
+	}
+	id := r.Header.Get("x-worker-id")
```

Comment:
Also let's prefix `x-webhooktunnel-` instead of just `x-` for all custom headers...